src/networking.c

```diff
@@ -4400,15 +4401,15 @@ int handleClientsWithPendingWritesUsingThreads(void) {
             continue;
         }
 
-        int target_id = item_id % server.io_threads_num;
+        int target_id = item_id % server.io_threads_active_num;
```

Comment:
it will be ideal to reduce the dependency on global vars such as `server.io_threads_active_num`. since `adjustIOThreadCount()` already returns the number of active thread count, we could just use its return value. 

```suggestion
        int target_id = item_id % io_threads_active_num;
```